switchbot/devices/curtain.py

```diff
@@ -168,8 +148,8 @@ def _update_motion_direction(
             self._is_closing = new_position < previous_position
 
     async def get_extended_info_summary(self) -> dict[str, Any] | None:
```

Comment:
I purposely didn't add unit tests for this method because it smells wrong. The code that checks `data[2]` but then only uses `data[1]` just seems wrong, but I don't have any paired curtain devices to test. I don't want to add test cases for code I suspect might be incorrect (but I can't test).

Reply:
That does look wrong.  I guess we never call this in HA?

Reply:
Doesn't look like `get_extended_info_summary` is used at all by HA

Reply:
https://github.com/OpenWonderLabs/SwitchBotAPI-BLE/blob/latest/devicetypes/curtain.md seems to confirm this is a bug so I'm going to fix that too. Not used by HA, but still a bug in the library.